DAP-941: Worker and copyWorker functions (#301)
bcgov/citz-grs-dats: README.md

## Quick Start

1. **Set Up Environment**: Set up the `.env` based on the `.env.template` file.

2. **Start Services**: Run `npm run up` to start the API, MongoDB, and RabbitMQ services.

3. **Desktop Directory**: Navigate to `desktop`.

4. **Install Dependencies**: Run `npm install`.

> 5. **Build App**: Run `npm run build` to enable worker functionality.

Reviewer: add note on running as admin based on os?

Author: Great idea!

Author: Actually now that I think about it I'm not sure if it's necessary. On Windows I can run `npm run build` without Admin privs, but not `npm run build:win`.

`npm run build` just creates the `out` directory and not the `dist`. Is it the same on Mac?

6. **Launch App**: Run `npm run dev`.

### Next Steps

- [Change API URL](#change-api-url)
- [Test Desktop App Executable](#test-desktop-app-executable)

<br />

## Change API URL

To switch API environments:

1. In the app, go to `Edit > Select API URL`.

2. Choose `Local`, `Dev`, `Test`, or `Prod`.
  - Default: `Local` during development using `npm run dev`, `Prod` in executable builds.

<br />

## Test Desktop App Executable

1. **Clear Build Folders**: Remove `desktop/out/` and `desktop/dist/`.

2. **Build Executable**:

  - Run from `desktop/` as an Administrator:

```
npm run build:win
npm run build:mac
npm run build:linux
```

> 3. **Uninstall Previous Version**: Remove existing installation (e.g., in `C:\Users\<username>\AppData\Local\Programs\desktop` on Windows).
> 
> 4. **Locate Build**: In VSCode, right-click `desktop/dist/` and select `Reveal in File Explorer`. Run the setup executable.
> 
> 5. **Test App Launch**: Ensure the desktop app opens correctly, then close it.

Reviewer: ```suggestion
3. **Uninstall Previous Version**: Remove existing installation
    a. Windows: in `C:\Users\<username>\AppData\Local\Programs\desktop`.
    b. Mac: `/Users/<username>/Applications/desktop.app`.

4. **Locate Build**: In VSCode, right-click `desktop/dist/` and select `Reveal in Finder/File Explorer`. Run the setup executable.

5. **Test App Launch**: Ensure the desktop app opens correctly, then close it.
```

> 6. **Run in Console**: Open Command Prompt and run `<file-location>/desktop.exe` (e.g., `C:\Users\<username>\AppData\Local\Programs\desktop\desktop.exe`). Execution may differ on Mac, use Terminal to run the executable.

Reviewer: ```suggestion
6. **Run in Console**:
    a. Windows: Open Command Prompt and run `<file-location>/desktop.exe`
        e.g., `C:\Users\<username>\AppData\Local\Programs\desktop\desktop.exe`
    b. Mac: Open Terminal and run `open <file location>`
        e.g., `open /Applications/desktop.app`
```

### Next Steps

- [Change API URL](#change-api-url)
- [Test Desktop App Executable](#test-desktop-app-executable)

<br />

## Change API URL

To switch API environments:

1. In the app, go to `Edit > Select API URL`.

2. Choose `Local`, `Dev`, `Test`, or `Prod`.
  - Default: `Local` during development using `npm run dev`, `Prod` in executable builds.

<br />

## Test Desktop App Executable

1. **Clear Build Folders**: Remove `desktop/out/` and `desktop/dist/`.

2. **Build Executable**:

  - Run from `desktop/` as an Administrator:

```
npm run build:win
npm run build:mac
npm run build:linux
```

3. **Uninstall Previous Version**: Remove existing installation
    a. Windows: in `C:\Users\<username>\AppData\Local\Programs\desktop`.
    b. Mac: `/Applications/desktop.app`.

4. **Locate Build**: In VSCode, right-click `desktop/dist/` and select `Reveal in Finder/File Explorer`. Run the setup executable.

5. **Test App Launch**: Ensure the desktop app opens correctly, then close it.

6. **Run in Console**:
    a. Windows: Open Command Prompt and run `<file-location>/desktop.exe`
        e.g., `C:\Users\<username>\AppData\Local\Programs\desktop\desktop.exe`

>     b. Mac: Open Terminal and run `open <file location>`
>         e.g., `open /Applications/desktop.app`

Reviewer: ```suggestion
    b. Mac: Open Terminal and run `<file location>`
        e.g., `/Applications/desktop.app/Contents/MacOS`
```
